lldb/source/Symbol/SwiftASTContext.cpp

```diff
@@ -1536,14 +1536,18 @@ static StringRef GetXcodeContentsPath() {
           std::chrono::seconds(
               3)); // Timeout in seconds to wait for shell program to finish
       if (status == 0 && !output.empty()) {
-        size_t first_non_newline = output.find_last_not_of("\r\n");
-        if (first_non_newline != std::string::npos) {
-          output.erase(first_non_newline + 1);
-        }
-
         size_t pos = output.rfind(substr);
         if (pos != std::string::npos) {
+          // Remove everything after the substring.
           output.erase(pos + strlen(substr));
+
+          // Strip any leading newline characters and everything before them.
+          // This needs to happen *after* the erase above because otherwise a
+          // trailing newline might erase the whole string.
+          size_t last_newline = output.find_last_of("\r\n");
```

Comment:
I don't see why the `\r` is necessary here. We are not going to run this on macOS < 10.0. On Windows `\r\n` will still contain `\n`.

Reply:
Shouldn't/Isn't this function in lldb/source/Plugins/Platform/MacOSX/PlatformMacOSX.cpp?

Reply:
ConstString PlatformMacOSX::GetSDKDirectory(lldb_private::Target &target) {

Reply:
Good point, this function is doing something different but it's also calling `xcrun`. Let me factor that out and reuse it in both places.